src/config/field-overrides/programStage.js

```diff
@@ -25,8 +25,27 @@ const reportDateOptions = [
  *  or translateOptions through fieldOptions to the DropDown-component.
  */
 
-const ReportDateToUseDropDown = props =>
-    <DropDown {...props} options={reportDateOptions} translateOptions />;
+const ReportDateToUseDropDown = props => (
+    <DropDown {...props} options={reportDateOptions} translateOptions />
+);
+
+const NextScheduleDateField = props => {
+    const dataElementsOptions = props.model.programStageDataElements
+        .filter(psde => psde.dataElement.valueType === 'DATE')
+        .map(psde => ({
+            text: psde.dataElement.displayName,
+            value: psde.dataElement,
+        }));
+
+    let value = undefined;
+    if (props.model.nextScheduleDate) {
+        value = dataElementsOptions
+            .filter(opts => opts.value.id === props.model.nextScheduleDate.id)
+            .map(v => v.value)[0];
+    }
```

Comment:
`.find` would be a better fit here:
```javascript
value = dataElementsOptions.find(opts => opts.value.id === props.model.nextScheduleDate.id).value
```

Reply:
I used filter because then I wouldn't need an extra check to check if value exists. I like that I can just map it, instead of accessing props directly on something that could be undefined. `[][0]` results in undefined and not an error.

Reply:
Yeah, true.... you'd have do do it it in two steps... In the future I'd suggest using the `.find` and using an `optional-chaining-operator`...

For now, I would still prefer to go with `.find` and use multiple constants:
```javascript
const selectedDEO =
    props.model.nextScheduleDate &&
    dataElementsOptions.find(
        opts => opts.value.id === props.model.nextScheduleDate.id
    )
const value = selectedDEO ? selectedDEO.value : undefined
```
Writing it like above is more readable and takes the same amount of lines as the original.

Reply:
Actually... 
`const value = selectedDEO && selectedDEO.value`
Would probably work too, since the value of an unfound `selectedDEO` should be `undefined`, and so would `props.model.nextScheduleDate`... And `undefined && undefined === undefined`